packages/cursorless-vscode/src/keyboard/TokenTypes.ts

```diff
@@ -0,0 +1,72 @@
+import { SimpleScopeTypeType } from "@cursorless/common";
+import { HatColor, HatShape } from "../ide/vscode/hatStyles.types";
+import {
+  KeyboardActionType,
+  SimpleKeyboardActionType,
+} from "./KeyboardActionType";
+
+/**
+ * Maps from modal keyboard config section name to the type of entry expected in
+ * that section.
+ */
+export interface SectionTypes {
+  actions: KeyboardActionType;
+  colors: HatColor;
+  misc: MiscValue;
```

Comment:
the awkward state is where you have a lot of structure and also a misc. it is hard for people to predict what goes in misc, and it is hard to refactor misc because backwards compat. I'd be pretty comfortable with dropping it all in one giant blob, or in having lots of structure; the concern is mainly around the mix.

Reply:
One drawback with giant blob for this specific case is that then they have no way to have two identifiers with the same key. If they can never both simultaneously be valid tokens, then it's ok for them to do that. Tho I guess they could accomplish the same thing by giving them each a different prefix and letting suffix magic shorten them. Idk

Reply:
> can't decide whether "giving them each a different prefix and letting suffix magic shorten them" is elegant or horrible.

I feel exactly the same way 😅

> i don't feel strongly. maybe let's leave as-is and then revisit as needed?

Yeah this is why I'm leaving all the new fancy stuff in this PR out of the user docs, so we can iterate quickly without worrying about backwards compatibility